packages/server/src/api/controllers/query/index.ts

```diff
@@ -163,38 +169,42 @@ export async function preview(ctx: UserCtx) {
     }
 
     const { rows, keys, info, extra } = (await Runner.run(inputs)) as any
-    const schemaFields: any = {}
+    const previewSchema: Record<string, QuerySchema> = {}
+    const makeQuerySchema = (type: FieldType, name: string): QuerySchema => ({
+      type,
+      name,
+    })
     if (rows?.length > 0) {
       for (let key of [...new Set(keys)] as string[]) {
         const field = rows[0][key]
         let type = typeof field,
-          fieldType = FieldTypes.STRING
+          fieldMetadata = makeQuerySchema(FieldType.STRING, key)
```

Comment:
NAB: Why use a function here? Why not:

```ts
let fieldMetadata: QuerySchema
...
fieldMetadata = {
  type: FieldType.STRING,
  name: key
}
```

Reply:
Just for clarity sake, reduces the chance of typos/extra props being added, Typescript does protect against this, but its just a rule I follow since we work in both JS/TS (with the frontend).

Reply:
Ah the extra props is a good point. Kind of like constructor overloading; create an object with the specified parameters only.